hikari/files.py

```diff
@@ -473,6 +473,38 @@ async def read(
         async with self.stream(executor=executor) as reader:
             return await reader.read()
 
+    async def save(
+        self,
+        *,
+        path: typing.Optional[typing.Union[pathlib.Path, str]] = None,
+        executor: typing.Optional[concurrent.futures.Executor] = None,
+    ) -> None:
+        """Save this resource to disk.
+
+        This writes the resource file in chunks, and so does not load
+        the entire resource into memory.
+
+        Parameters
+        ----------
+        path : typing.Optional[typing.Union[pathlib.Path, str]]
+            The path to save this resource to. If this is a string, the
+            path will be relative to the current working directory. If
+            `builtins.None`, the resource will be saved in the
+            current working directory with the same name as it has on
+            Discord.
+        executor : typing.Optional[concurrent.futures.Executor]
+            The executor to run in for blocking operations.
+            If `builtins.None`, then the default executor is used for the
+            current event loop.
+        """
+        loop = asyncio.get_running_loop()
+        path = path or self.filename
```

Comment:
I'm tempted to add some things here to (1) ensure the same filetype, and (2) account for directories being passed.

Reply:
If a directory is passed the filename will need to be appended, no?

Reply:
Oh, that's what you mean. Yeah, you could make that an option. Should be easy enough with pathlib